src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -18,221 +18,76 @@ namespace System
+        /// (i.e. destination.Length &lt; source.Length).
         /// </exception>
-        public static void PadLeft(this ReadOnlySpan<char> source, int totalWidth, Span<char> destination)
+        public static void PadLeft(this ReadOnlySpan<char> source, Span<char> destination)
         {
-            if (totalWidth < 0)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
-            if (destination.Length < totalWidth)
-                ThrowHelper.ThrowArgumentException_DestinationTooShort();
-
-            int count = totalWidth - source.Length;
-            if (count <= 0)
-            {
-                // This will throw if destination.Length < source.Length
-                source.CopyTo(destination);
-            }
-            else
-            {
-                source.CopyTo(destination.Slice(count));
-                for (int i = 0; i < count; i++)
-                    destination[i] = ' ';
-            }
+            source.PadLeft(destination, ' ');
         }
 
         /// <summary>
-        /// Moves the source characters to the destination and right-aligns them by padding with 
-        /// a specified Unicode character on the left, for a specified total length.
+        /// Copies the source characters to the destination and right-aligns them by padding with 
+        /// a specified Unicode character on the left, until the destination is filled.
         /// </summary>
         /// <param name="source">The source span that contains the characters to be padded.</param>
-        /// <param name="totalWidth">The number of characters in the resulting span, 
-        /// equal to the number of original characters plus any additional padding characters.</param>
         /// <param name="destination">The span that contains the resulting right-aligned and padded characters.</param>
         /// <param name="paddingChar">The Unicode characters used for the padding.</param>
-        /// <remarks>
-        /// If the totalWidth is less than or equal to the length of the source,
-        /// the entire source is copied into the destination without padding.
-        /// </remarks>
         /// <exception cref="System.ArgumentException">
-        /// Thrown when the destination span is too small to contain the source span with the padding
-        /// (i.e. destination.Length &lt; totalWidth or  destination.Length &lt; source.Length).
+        /// Thrown when the destination span is too small to contain the source span to be padded
+        /// (i.e. destination.Length &lt; source.Length).
         /// </exception>
-        /// <exception cref="System.ArgumentOutOfRangeException">
-        /// Thrown when <paramref name="totalWidth"/> is less than zero.
-        /// </exception>
-        public static void PadLeft(this ReadOnlySpan<char> source, int totalWidth, Span<char> destination, char paddingChar)
+        public static void PadLeft(this ReadOnlySpan<char> source, Span<char> destination, char paddingChar)
```

Comment:
Do we feel this (and similar) methods are useful? They are literally just copy and fill.
@stephentoub, I would be interested especially in your opinion as you requested these (string-like) APIs after trying to port code from string to 'Span<char>', but I am not sure which of these new members would actually help in the port.

Reply:
> I would be interested especially in your opinion as you requested these (string-like) APIs after trying to port code from string to 'Span', but I am not sure which of these new members would actually help in the port.

To me, the most important string-like methods we can provide are those for comparison/equality.  Doing it correctly in the face of cultures is challenging outside of the platform and requires a dependency on the OS or a library like ICU.  Even just for ordinal it's difficult to do efficiently, with optional handling of casing.  This is my Pri0, methods like Compare, Equals, StartsWith, etc.

My Pri1 is support for trimming and searching, methods like Trim and IndexOf.  These are possible for a dev to write, handfuls of lines of code, but they're so commonly used, I believe it's important that we have built-in support for these that we can implement once and tune and make available for all.

You can see at https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/StringSpanHelpers.cs the APIs I needed in support of the span work in corelib itself; the methods I mentioned above account for almost all of the methods in that file.  The only other one is Remove, which I concede is an oddity due to it needing to allocate in some situations.  Then again, the implementation in that file is obviously missing some special cases where it wouldn't need to allocate (e.g. where it's just removing from the beginning or end), which would necessitate more code, which highlights that even something like that can be useful for us to provide.

As for PadLeft/PadRight, I can probably count on two hands the number of times I've used those even with strings, so they're not high on my priority list :)

Reply:
> This is my Pri0, methods like Compare, Equals, StartsWith, etc.

Working on these next :)

Reply:
@terrajobst, @jkotas, @joshfree, @stephentoub, should we remove the Pad APIs too? What are your thoughts?

Reply:
> should we remove the Pad APIs too? What are your thoughts?

Let's remove them for now.  We can add them in the future if we find a real need.

Reply:
I agree with @stephentoub

Reply:
OK. Updated the PR.